### PR TITLE
2.0.0 with tests

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [x] Ensured the license file is being packaged.

Carries on with #2 (which would be dangerous to merge due to the dependency change):
- fix `graphql-core >=2.2,<3`
- add tests (based on github tag)
- remove vendored license (now included in tarball)
- generally modernize recipe
- add self (@bollwyvl) as maintainer 
  > @carlodri: interested in co-maintainership?

### CODE REVIEW

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by carlodri** [comment]
@bollwyvl thanks for the ping, I don't know if I can be of much help (I'm not a graphql user myself), so I would skip this one. However, if you want a maintenance pal, I'm happy to be thrown in!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,46 +1,49 @@
...
   version: {{ version }}
 
 source:
-  fn: {{ name }}-{{ version }}.{{ bundle }}
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ bundle }}
-  {{ hash_type }}: {{ hash }}
+  - url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+    sha256: 7fa74661246e826ef939ee92e768f698df167a7617361ab399901eaebf80dce6
+    folder: dist
+  - url: https://github.com/graphql-python/{{ name }}-py/archive/v{{ version }}.tar.gz
```
Comment (carlodri) [inline recipe/meta.yaml]: why do you need to manually pull this in?

**Comment by bollwyvl** [comment]
Bringing in the src because the tests are not distributed in the pypi
tarball, and it seemed like getting some tests working would help prevent
putting out bad builds while they're bumping make revs all over the chain.

**Comment by bollwyvl** [comment]
And if you're not looking for more chores, @carlodri, I totally understand!

**Comment by carlodri** [comment]
ok I see your point on tests, sure!

As for the maintenance, yes, I will skip this one for the moment, thanks!

File: recipe/meta.yaml (hunk 1 above)
Comment (mariusvniekerk) [reply recipe/meta.yaml]: So the one thing that this might do is confuse the bot :(  You may want to ONLY use the GH tarball in this case

Comment (carlodri) [reply recipe/meta.yaml]: @bollwyvl are you able to refactor as per @mariusvniekerk's recommendation? If you give me access to your fork, I can also try to do it.

Comment (bollwyvl) [reply recipe/meta.yaml]: cheers, sorry i missed that feedback. updated to just use the tagged github tarball.

**Comment by carlodri** [comment]
@pmlandwehr can you please merge? This is blocking updates downstream. Thanks!

**Pull request merged**